### PR TITLE
move to go 1.19, latest github actions, and golangci-lint

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

**Comment by benkeith-splunk** [comment]
Looks fine except for the one remaining lint issue.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -34,7 +34,4 @@ jobs:
         run: |
           go mod download
           set -x
-          go test -covermode atomic -race -timeout 120s -coverprofile ./coverage.out $(go list ./...)
-          cov="$(go tool cover -func coverage.out | grep -v 100.0% || true)"
-          echo $cov
-          test -z "$cov"
\ No newline at end of file
+          go test -race -timeout 120s $(go list ./...)
```
Comment (jgheewala) [inline .github/workflows/test.yml:37]: why would you remove code coverage check? The goal is to always keep it at 100%

Comment (atoulme) [reply .github/workflows/test.yml:37]: It currently fails and blocks work. We can reestablish it in a separate PR once the build passes again.

Comment (atoulme) [reply .github/workflows/test.yml:37]: Initially I meant for this PR to be lean and mean, but given that it now changes every other file, I can look into adding coverage if I get around to it.

**Review by jgheewala — APPROVED** [review]
lgtm only thing i am worried about is removing the 100% coverage check.

File: .github/workflows/test.yml (hunk 1 above)
Comment (atoulme) [reply .github/workflows/test.yml]: alright it's back on and we're passing the build. Merging.

**Pull request merged**